### PR TITLE
Add AWS Terraform template

### PR DESCRIPTION
**Pull request by clbarrell**
This PR drops in a CI template for applying AWS changes using Terraform. 

It's mostly based on the workflow outlined in https://developer.hashicorp.com/terraform/tutorials/automation/automate-terraform

This could be generalised down to terraform (minus AWS) if we wanted to removed OIDC part, which I think seems reasonable.

Here's an example: 
Screenshot 2024-02-05 at 10 21 07 am

### CODE REVIEW

**Comment by clbarrell** [comment]
Realised I already had made this one: https://github.com/buildkite/templates/pull/23

File: terraform-ci/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+steps:
+  - label: ":terraform: :mag: validate"
+    plugins:
+      - docker#v5.9.0:
+          image: hashicorp/terraform:1.5
+          command: ["validate"]
+          environment:
```
Comment (dannymidnight) [inline terraform-ci/pipeline.yaml:7]: We can clean this up a bit with the following flag:

https://github.com/buildkite-plugins/docker-buildkite-plugin?tab=readme-ov-file#propagate-aws-auth-tokens-optional-boolean

File: terraform-ci/pipeline.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+steps:
+  - label: ":terraform: :mag: validate"
+    plugins:
+      - docker#v5.9.0:
+          image: hashicorp/terraform:1.5
+          command: ["validate"]
```
Comment (dannymidnight) [inline terraform-ci/pipeline.yaml:6]: Personal preference here, I prefer the command to be included at the top level as opposed to the docker command.  Makes it a smidge easier to refactor and reads a little better IMO.

**Review by dannymidnight — COMMENTED** [review]
Nice one. I think we should do something similar to the pulumi pipeline with a block step for deploying the changes. 

https://github.com/buildkite/templates/blob/main/pulumi-ci-cd/pipeline.yaml

**Pull request merged**